src/torchsim/models/mprage.py

```diff
@@ -140,16 +140,16 @@ def _engine(
         states = epg.adiabatic_inversion(states, inv_efficiency)
         states = epg.longitudinal_relaxation(states, E1inv, rE1inv)
         states = epg.spoil(states)
-
+        signal = []
```

Comment:
If we want to store magnetization for each spgr shot (e.g., to study signal modulation in k-space or for contrast-resolved reconstruction), it is better to use [torchsim.models.MPnRAGEModel](https://github.com/INFN-MRI/torchsim/blob/main/src/torchsim/models/mpnrage.py).